test/reducer.test.js

```diff
@@ -92,24 +114,60 @@ describe('OFFLINE_ACTION action type', () => {
       it('actions are pushed into the queue in order of arrival', () => {
         const preAction = actionCreators.connectionChange(false);
         const action1 = actionCreators.fetchOfflineMode(prevActionToRetry1);
-        const prevState = reducer(initialState, preAction);
+        const prevState = createReducer()(initialState, preAction);
 
-        let nextState = reducer(prevState, action1);
+        let nextState = createReducer()(prevState, action1);
 
         expect(nextState).toEqual({
           isConnected: false,
           actionQueue: [prevActionToRetry1],
         });
 
         const action2 = actionCreators.fetchOfflineMode(prevActionToRetry2);
-        nextState = reducer(nextState, action2);
+        nextState = createReducer()(nextState, action2);
 
         expect(nextState).toEqual(
           getState(false, prevActionToRetry1, prevActionToRetry2),
         );
       });
     });
 
+    describe('thunks that are the same with custom comparison function', () => {
+      const thunkFactory = (id, name, age) => {
+        function thunk(dispatch) {
+          dispatch({ type: 'UPDATE_DATA_REQUEST', payload: { id, name, age } });
+        }
+        thunk.meta = {
+          args: { id, name, age },
+        };
+        return thunk;
+      };
+
+      it(`should add thunks if function is same but thunks are modifying different items`, () => {
+        const prevState = getState(false, thunkFactory(1, 'Bilbo', 55));
+        const thunk = actionCreators.fetchOfflineMode(
+          thunkFactory(2, 'Link', 54),
+        );
+        const nextState = createReducer(comparisonFn)(prevState, thunk);
+
+        expect(nextState).not.toEqual(
```

Comment:
Better to test here that we have indeed two items on the queue and their shape is what we expect?